lib/Graph/Nodes.cpp

```diff
@@ -228,8 +228,11 @@ static void verifyTanh(NodeValue src, NodeValue dest) {
 }
 
 static void verifySoftMax(NodeValue src, NodeValue dest) {
-  checkSameType(src, dest);
-  assert(src.dims() == dest.dims() && "Invalid shape");
+  if (src.getType()->isQuantizedType()) {
+    checkSameShape(src, dest);
```

Comment:
nit: Did you decide against also verifying src and dest have the same element type? Why not have it here?

Reply:
@jfix71 I actually did not add the element type equality check on purpose. For quantized networks, I think in general that we can't necessarily force the output element type to be the same as the input element type. I think this choice should be left to the quantization algorithm and it can depend on the target.

Reply:
Not testing the element type seems to be controversial and since I don't want to delay too much this PR that is primarily about solving a scale/offset issue, I reintroduced the element type checking for softmax. I'm currently doing some experiments about quantization and we can re-discuss more globally this element type problematic later in a different PR or issue.
As a side note, I've noticed that U8 is now handled in Glow not by introducing a new element type, but by acting on the S8 offset, so checking the element type equality between source and destination of nodes should not cause any issue in case the input and the output types of a node don't have the same signedness.